src/mlpack/methods/ann/layer/concat_impl.hpp

```diff
@@ -171,11 +164,7 @@ void ConcatType<MatType>::Backward(
     slices *= this->outputDimensions[i];
 
   arma::Cube<typename MatType::elem_type> gyTmp;
-  MakeAlias(gyTmp,
-            (typename MatType::elem_type*) gy.memptr(),
-            rows,
-            this->outputDimensions[axis],
-            slices);
+  MakeAlias(gyTmp, gy, rows, this->outputDimensions[axis], slices);
```

Comment:
I left another note on the SFINAE bits, but I *think* this is calling the overload of `MakeAlias` that is for matrices, since `gy` is an `arma::Mat` (not an `arma::Cube`), and taking `offset = slices`, which is not the intention.